orttraining/orttraining/python/training/ortmodule/ortmodule.py

```diff
@@ -16,28 +16,31 @@
 
 class ORTModule(torch.nn.Module):
-    """Specializes a user torch.nn.Module to leverage ONNX Runtime graph execution.
-
-    ORTModule specializes the user's torch.nn.Module and provides forward, backward
-    implementations be leveraging ONNX Runtime.
-
-    ORTModule interacts with:
-    - GraphExecutionManagerFactory: Which returns a GraphExecutionManager based on
-    whether or not the user's torch module is in training mode or eval mode.
-    - GraphExecutionManager: Responsible for building and executing the forward and backward graphs.
-        - InferenceManager(GraphExecutionManager): Responsible for building, optimizing
-        and executing the inference onnx graph.
-        - TrainingManager(GraphExecutionManager): Responsible for building, optimizing
-        and executing the training onnx graph.
-
-        The GraphExecutionManager first exports the user model into an onnx model.
-        Following that, GraphExecutionManager interacts with OrtModuleGraphBuilder to optimize the onnx graph.
-        Once the onnx graph has been optimized, an ExecutionAgent is instantiated that
-        facilitates in executing the forward and backward subgraphs of the onnx model.
-
-    - _ortmodule_io: Provides utilities to transform the user inputs and outputs of the model.
-        - It facilitates in flattening the output from the user's PyTorch model (since exporting
-        of nested structures is not supported at the moment)
+    """Extends user's :class:`torch.nn.Module` model to leverage ONNX Runtime super fast training engine.
+
+    ORTModule specializes the user's :class:`torch.nn.Module` model, providing :meth:`~torch.nn.Module.forward`,
+    :meth:`~torch.nn.Module.backward` along with all others :class:`torch.nn.Module` APIs leveraging ONNX Runtime.
+
+    ORTModule interacts with
+
+    * | :class:`GraphExecutionManagerFactory` which returns a :class:`GraphExecutionManager` based on
+      | whether or not the user's torch module is in training mode or eval mode
+
+    * :class:`GraphExecutionManager` builds and execute the forward/backward graphs
+
+        * | :class:`InferenceManager(GraphExecutionManager)` builds, optimizes and execute the inference ONNX graph
+
+        * | :class:`TrainingManager(GraphExecutionManager)` builds, optimizes and executes the
+          | training onnx graph. The :class:`GraphExecutionManager` first exports the user model into an ONNX model.
+          | Next, :class:`GraphExecutionManager` interacts with :class:`OrtModuleGraphBuilder` to optimize it.
+          | Finally, :class:`ExecutionAgent` is instantiated using the optimized ONNX graph to
+          | facilitate execution of the forward/backward subgraphs of the ONNX model
+
+    * py:module:`_io` provides utilities to transform the user inputs and outputs for the model
+
+        * | It facilitates in flattening the output from the user's PyTorch model (since exporting
+          | of nested structures is not supported at the moment)
```

Comment:
This is useful developer-focused documentation, but I don't think we should be describing implementation details (and private APIs) as part of our public docs. Can you move this somewhere else?